examples/util/WebNNRunner.js

```diff
@@ -0,0 +1,427 @@
+          throw new Error(`Unrecognized model format, support TFLite | ONNX | OpenVINO model`);
+      }
+    } else {
+      throw new Error(`There's none model file info, please check config info of modelZoo.`);
+    }
+
+    this._setRawModel(rawModel);
+    this._setLoadedFlag(true);
+    return status;
+  };
+
+  /**
+   * This method is to load model with given model info.
+   * @param modelInfo: A string for model info.
+   * @returns {string} This returns 'LOADED' status if model already loaded.
+   */
+  loadModel = async (modelInfo) => {
+    if (this._bLoaded && this._currentModelInfo.modelFile === modelInfo.modelFile) {
+      return 'LOADED';
+    }
+
+    // reset all states
+    this._setLoadedFlag(false);
+    this._setInitializedFlag(false);
+    this._setBackend(null);
+    this._setPrefer(null);
+    this._setModelInfo(modelInfo);
+    this._setModelRequiredOps(new Set());
+    this._setDeQuantizeParams([]);
+    this._setSubgraphsSummary([]);
+    this._initInputTensor();
+    this._initOutputTensor();
+
+    await this._loadModelFile(this._currentModelInfo.modelFile);
+
+    if (this._currentModelInfo.labelsFile != null) {
+      await this._loadLabelsFile(this._currentModelInfo.labelsFile);
+    }
+  };
+
+  /**
+   * This method is to get typedArray type for inputTensor.
+   * @returns {object} This returns Uint8Array or Float32Array object or other typedArray type if inherited.
+   */
+  _getInputTensorTypedArray = () => {
+    // Override by inherited if needed
+    const typedArray = this._currentModelInfo.isQuantized || false ? Uint8Array : Float32Array;
+    return typedArray;
+  };
+
+  /**
+   * This method is to init set '_inputTensor'.
+   */
+  _initInputTensor = () => {
+    const typedArray = this._getInputTensorTypedArray();
+    this._inputTensor = [new typedArray(this._currentModelInfo.inputSize.reduce((a, b) => a * b))];
+  };
+
+  /**
+   * This method is to typedArray type for outputTensor.
+   * @returns {object} This returns Uint8Array or Float32Array object or other typedArray type if inherited.
+   */
+  _getOutputTensorTypedArray = () => {
+    // Override by inherited if needed
+    const typedArray = this._currentModelInfo.isQuantized || false ? Uint8Array : Float32Array;
+    return typedArray;
+  };
+
+  /**
+   * This method is to init set '_outputTensor'.
+   */
+  _initOutputTensor = () => {
+    // Override by inherited if needed
+    const typedArray = this._getOutputTensorTypedArray();
+    const outputSize = this._currentModelInfo.outputSize;
+
+    if (typeof outputSize === 'number') {
+      this._outputTensor = [new typedArray(outputSize)];
+    } else {
+      this._outputTensor = [new typedArray(outputSize.reduce((a, b) => a * b))];
+    }
+  };
+
+  /**
+   * This method is to compile a machine learning model by backend and prefer.
+   * @param options: A string has backend and prefer info.
+   * @returns {string} This returns a string for status of compilation model.
```

Comment:
If the status represents the status of the runner, please keep it as a property not other than return it as value.